cms/djangoapps/contentstore/signals/handlers.py

```diff
@@ -52,23 +49,10 @@ def listen_for_course_publish(sender, course_key, **kwargs):  # pylint: disable=
     registering proctored exams, building up credit requirements, and performing
     search indexing
     """
-    is_enabled = ENABLE_ASYNC_REGISTER_EXAMS.is_enabled(course_key)
-    if is_enabled:
-        from cms.djangoapps.contentstore.tasks import update_special_exams_and_publish
-        course_key_str = str(course_key)
-        update_special_exams_and_publish.delay(course_key_str)
-    else:
-        # first is to registered exams, the credit subsystem will assume that
-        # all proctored exams have already been registered, so we have to do that first
-        try:
-            register_special_exams(course_key)
-        # pylint: disable=broad-except
-        except Exception as exception:
-            log.exception(exception)
-
-        # then call into the credit subsystem (in /openedx/djangoapps/credit)
-        # to perform any 'on_publish' workflow
-        on_course_publish(course_key)
+    # register special exams asynchronously
+    from cms.djangoapps.contentstore.tasks import update_special_exams_and_publish
```

Comment:
I think this could be simplified and grouped into the imports at line 58. If the imports are grouped, you could also replace other references to str(course_key) in lines 61 and 66 with the var course_key_str